3scale/porta
# Fix spam protection in the reset password screen

**What this PR does / why we need it**:

THREESCALE-7431 was not only about fixing the spam protection in the signup screen: the reset password screen had a broken recaptcha as well. The same fix should be good for both screens but the controller for the reset password form wasn't taking the spam check params correctly, so the checks always failed and the activity was always being marked as suspicious. Fixed in https://github.com/3scale/porta/pull/3215/files#diff-811e0b68328f92343b613fa4b352676067bc7e9f71fe951f627de512d2a21b62R49-R51

I also added some cukes to the reset password screen to test this behavior.

**Which issue(s) this PR fixes** 

[THREESCALE-7431](https://issues.redhat.com/browse/THREESCALE-7431)

**Verification steps** 

1. Fill in the reset password form in a non-suspicious way: Javascript enabled, correct email, and take a few seconds to fill it in.
2. Click "Send instructions" 
3. The spam check should pass, so you shouldn't see the captcha.
